gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -105,17 +105,12 @@ public void setText (BitmapFont font, CharSequence str, int start, int end, Colo
 			return;
 		}
 
-		if (truncate != null)
-			wrap = true; // Ensures truncate code runs, doesn't actually cause wrapping.
-		else if (targetWidth <= fontData.spaceXadvance * 3) //
-			wrap = false; // Avoid one line per character, which is very inefficient.
+		float minimumTargetWidthForWrapping = fontData.spaceXadvance * 3; // Avoid one line per character, which is very inefficient.
+		boolean wrapOrTruncate = (wrap && targetWidth > minimumTargetWidthForWrapping) || truncate != null;
```

Comment:
@NathanSweet @tommyettinger I'm not really sure what to do, please change at will